be/src/exec/pipeline/runtime_filter_types.h

```diff
@@ -315,16 +316,23 @@ class PartialRuntimeFilterMerger {
         for (auto count : _ht_row_counts) {
             row_count += count;
         }
+
         for (auto& desc : _bloom_filter_descriptors) {
             desc->set_is_pipeline(true);
             // skip if it does not have consumer.
             if (!desc->has_consumer()) continue;
             // skip if ht.size() > limit, and it's only for local.
-            if (!desc->has_remote_targets() && row_count > _limit) continue;
+            if (!desc->has_remote_targets() && row_count > _local_rf_limit) continue;
             LogicalType build_type = desc->build_expr_type();
             JoinRuntimeFilter* filter = RuntimeFilterHelper::create_runtime_bloom_filter(_pool, build_type);
             if (filter == nullptr) continue;
-            filter->init(row_count);
+
+            if (desc->has_remote_targets() && row_count > _global_rf_limit) {
```

Comment:
The size of the global runtime filter is determined by FE. Why is this restriction still required?

Reply:
> The size of the global runtime filter is determined by FE. Why is this restriction still required?

 in sender side, all partial runtime filters need to be built and send to grf coordinator although their sizes already exceed global_runtime_filter_build_max_size, which is not necessary.

so we can give up building partial rf early

Reply:
@silverbullet233  The planner in FE will not generate RF in the plan, if the estimated caldinality of the right table is greater than `global_runtime_filter_build_max_size`. What's the case will benefit from this restriction in BE? maybe when the estimated caldinality is biased?

Reply:
> @silverbullet233 The planner in FE will not generate RF in the plan, if the estimated caldinality of the right table is greater than `global_runtime_filter_build_max_size`. Which case will benefit from this restriction in BE? maybe when the estimated caldinality is biased?

yes, you are right.